### PR TITLE
Initial work on middleware datasource

### PR DESCRIPTION
**Pull request by psav**
# Purpose or Intent

@hhovsepy perhaps I am taking too much on here, there seems to be some intricate decision mapping. What I did do is create you 3 destinations which mimic the behaviour you have already implemented, it _should_ I hope simplify a few things. Please let me know if you like it, there is always the possibility of making the Details destination make hte decision on whether to use DetailsFromServer, if it has the server attribute by adding a conditional in the step function and doing something like. `self.navigate_obj.navigate_to(self, 'DetailsFromProvider')` etc

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 633852581ace78a8f35332c55172fed6a2a515a6:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by hhovsepy** [comment]
Fixed failing datasource tests in PR: https://github.com/ManageIQ/integration_tests/pull/3469

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/middleware/datasource.py [hunk 1]
```diff
@@ -97,8 +87,10 @@ class MiddlewareDatasource(MiddlewareBase, Taggable):
                        ('driver_name', 'driver_name'), ('jndi_name', 'jndi_name'),
                        ('connection_url', 'connection_url'), ('enabled', 'enabled')]
     taggable_type = 'MiddlewareDatasource'
+    appliance = CurrentAppliance()
 
-    def __init__(self, name, server, provider=None, **kwargs):
+    def __init__(self, name, server, provider=None, appliance=None, **kwargs):
+        self.appliance = appliance or get_or_create_current_appliance()
```
Comment (mfalesni) [inline cfme/middleware/datasource.py]: I guess this should be done the new way :stuck_out_tongue:

Comment (psav) [inline cfme/middleware/datasource.py]: ofc

**Comment by dajoRH** [comment]
Lint report for commit 79937515981bc1c59908e497f64f31c9c4b3616d:

`cfme/middleware/datasource.py`:
- :red_circle: Line 88:17: **F821** *undefined name 'CurrentAppliance'*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by psav** [comment]
Closing @hhovsepy this is staying around for you to have as a reference but will not be merged like it is.

**Pull request closed without merging**